lib/perl/Genome/Model/Tools/Validation/ProcessSomaticValidation.pm

```diff
@@ -751,30 +705,86 @@ sub execute {
       `grep -w Somatic $small_indel_file | cut -f 1-5 >$output_dir/$sample_name/indels/small.indels`;
       `cat $output_dir/$sample_name/indels/small.indels $output_dir/$sample_name/indels/large.indels | joinx sort >$output_dir/$sample_name/indels/indels.hq`;
   }
+  my $indel_file = "$output_dir/$sample_name/indels/indels.hq";
+  
 
-
-
+  #SVs
+  my $process_svs = $self->process_svs;
+  my $sv_file = "$build_dir/validation/sv/assembly_output.csv.merged.readcounts.somatic.wgs_readcounts.somatic";
+  if($process_svs){
+      unless( -e $sv_file ){
+          $self->warning_message("SV results for $sample_name not found, skipping SVs");
+          $process_svs = 0;
+      }
+  }
   if($process_svs){
       `mkdir $output_dir/$sample_name/svs`;
       `cp $sv_file $output_dir/$sample_name/svs/svs.hq` unless( -e "$output_dir/$sample_name/svs/$sv_file");
   }
 
-  #only need to operate on the non-zero files
-  @snv_files = ();
-  push(@snv_files,"$output_dir/$sample_name/snvs/snvs.validated") if (-s "$output_dir/$sample_name/snvs/snvs.validated");
-  push(@snv_files,"$output_dir/$sample_name/snvs/snvs.notvalidated") if (-s "$output_dir/$sample_name/snvs/snvs.notvalidated");
-  push(@snv_files, "$output_dir/$sample_name/snvs/" . basename($newcalls)) if (-s "$output_dir/$sample_name/snvs/" . basename($newcalls));
-
-  my $indel_file = "$output_dir/$sample_name/indels/indels.hq";
-
 
+  
   #clean the files
   my $i=0;
   for($i=0;$i<@snv_files;$i++){
       $snv_files[$i] = cleanFile($snv_files[$i])
   }
   $indel_file = cleanFile($indel_file);
 
+  #-------------------------------------------------
+  #filter out the off-target regions, if target regions are available
+  if($self->restrict_to_target_regions){
+      $self->status_message("Filtering out off-target regions...");
+      
+      my $featurelist;
+      if($self->target_regions) {
+          $featurelist = $self->target_regions;
+      } else { 
+          $featurelist = $model->target_region_set->file_path;
```

Comment:
It's a corner case, but it's possible for `$model` and `$build` to have a different `target_region_set` set.  Would it be better to prefer the `$build`'s set?